compiler/rustc_middle/src/ty/generics.rs

```diff
@@ -198,6 +198,21 @@ impl<'tcx> Generics {
             _ => bug!("expected const parameter, but found another generic parameter"),
         }
     }
+
+    /// Returns `true` if `params` has `impl Trait`.
+    pub fn has_impl_trait(&'tcx self) -> bool {
+        self.params.iter().any(|param| {
```

Comment:
I guess we could also add an `fn is_impl_trait` to `param`, but that's not necessary in this PR.